march_hardware/src/IMotionCube.cpp

```diff
@@ -266,6 +266,15 @@ float IMotionCube::getIMCVoltage()
          static_cast<float>(imc_voltage_iu);  // Conversion to Volt, see Technosoft CoE programming manual
 }
 
+float IMotionCube::getMotorVoltage()
+{
+  uint16_t motor_voltage = 0;
+  int val_size = 16;
+  sdo_bit32_write(slaveIndex, 0x2064, 0, 0x0232);
+  sdo_bit16_read(slaveIndex, 0x2066, 0, val_size, motor_voltage);
```

Comment:
Is it possible to read and write to SDOs, while ethercat is fully operational? I thought they were one-time messages, only sent at initialization. And since they are non-cyclic, won't it take a whole ethercat cycle to read out this SDO?

Reply:
I am not sure, @Deakensbubbles requested to use this method in one of my previous pull request so I thought it was fine to use this. But @Roelemans and @JorisWeeda are also concerned about this, so maybe we need to find another solution. But I don't know what to do, so maybe nice to discuss this with some people?

Reply:
I've taken the time to read a little bit more about the response time of an SDO request. I do think it is possible to request a value using this structure but the receive timeout stated in SOEM is 0.7 seconds for the SDO whereas our PDO timing is significantly less. I think a much more interesting problem is that the received value does not have a reliable timestamp because the delay can differ. Resulting in a value which may not be usable.